simple

```diff
@@ -69,26 +76,31 @@ create-template)
 ###############################################################################
 pre-compile)
     (
+	[ -e $cli_conf_dir ] || (
+	    mkdir $cli_conf_dir
+	) || exit
+  
 	[ -e $installed_pip ] || (
-	    mkdir -p $compiler_dir/.temp && cd $compiler_dir &&
+	    mkdir -p $compiler_dir/.temp &&
 	    yum install -y python-pip && pip install --upgrade pip &&
 	    touch $installed_pip
 	) || exit
-
-	cd $compiler_dir
-	pip install virtualenv || exit
-	virtualenv .env || exit
-	source ./.env/bin/activate
+	
+	[ -e $installed_virtualenv ] || (
+	    yum install -y python-virtualenv && touch $installed_virtualenv
+	) || exit
 
 	[ -e $installed_compiler ] || (
-	    pip install simple-grid-yaml-compiler && touch $installed_compiler
+	    virtualenv $compiler_env
+	    source $compiler_env/bin/activate && pip install simple-grid-yaml-compiler && touch $installed_compiler
 	) || exit
 
+	cd $compiler_dir
+	source $compiler_env/bin/activate
 	simple_grid_yaml_compiler $conf -o output.yaml -s schema.yaml
 	val=$?
 
 	deactivate
```

Comment:
Is that deactivate still needed right before the subshell exits?

Reply:
Ah I see your point. Maybe not. I can try without ;)

Reply:
Or must we not make any assumptions about what a deactivation may or may not do?

Reply:
Seems to work okay. I have removed it for now ;)